src/accessControl/acp/index.js

```diff
@@ -323,76 +330,49 @@ function getPolicyUrlName(policyUrl, name) {
+    };
+  }
+  const acr = acp.addAcrPolicyUrl(datasetWithAcr, policyUrl);
   return {
-    changed: changed || !existingAccessControl,
-    acr: existingAccessControl
-      ? datasetWithAcr
-      : chain(datasetWithAcr, (acr) => acpv1.addAcrPolicyUrl(acr, policyUrl)),
+    changed: changed || !existingPolicies,
+    acr,
   };
 }
 
 function ensureApplyMembers(policyUrl, datasetWithAcr, changed) {
-  const existingAccessControl = acpv2
-    .getMemberPolicyUrlAll(datasetWithAcr)
-    .find((url) => policyUrl === url);
-  if (existingAccessControl) {
+  const policies = acp.getMemberPolicyUrlAll(datasetWithAcr);
+  const existingPolicies = policies.find((url) => policyUrl === url);
+  if (existingPolicies) {
     return {
       changed,
       acr: datasetWithAcr,
     };
   }
-  let acr = datasetWithAcr;
-  try {
-    acr = acpv2.addMemberPolicyUrl(acr, policyUrl);
-  } catch (error) {
-    // TODO: Handle this error (probably by replacing acpv1.setControl with newer ACP APIs)
-    // Same problem as noted above, but this time it's acpv1.setControl that fails to handle
-    // datasetWithAcr properly. It doesn't seem to affect the outcome though, so we'll leave
-    // it like this for now
-  }
+  const acr = acp.addMemberPolicyUrl(datasetWithAcr, policyUrl);
   return {
-    changed: changed || !existingAccessControl,
+    changed: changed || !existingPolicies,
     acr,
   };
 }
 
 function ensureApplyControl(policyUrl, datasetWithAcr, changed) {
-  let accessControls = [];
-  try {
-    accessControls = acpv1.getAllControl(datasetWithAcr);
-  } catch (error) {
-    // TODO: Handle this error (probably by replacing acpv1.getAllControl with newer ACP APIs)
-    // For some reason the inner working of acpv1.getAllControl fails to run a getThingAll on the
-    // datasetWithAcr. It doesn't seem to affect the outcome though, so we'll leave it like this
-    // for now
-  }
-  const existingAccessControl = accessControls.find((ac) =>
-    acpv1.getPolicyUrlAll(ac).find((url) => policyUrl === url)
-  );
-  if (existingAccessControl) {
+  const policies = acp.getPolicyUrlAll(datasetWithAcr);
+  const existingPolicies = policies.find((url) => policyUrl === url);
+  if (existingPolicies) {
     return {
       changed,
       acr: datasetWithAcr,
     };
   }
-  let acr = datasetWithAcr;
-  try {
-    acr = acpv1.setControl(
-      datasetWithAcr,
-      chain(acpv1.createControl(), (ac) => acpv1.addPolicyUrl(ac, policyUrl))
-    );
-  } catch (error) {
-    // TODO: Handle this error (probably by replacing acpv1.setControl with newer ACP APIs)
-    // Same problem as noted above, but this time it's acpv1.setControl that fails to handle
-    // datasetWithAcr properly. It doesn't seem to affect the outcome though, so we'll leave
-    // it like this for now
-  }
+  const acr = acp.addPolicyUrl(datasetWithAcr, policyUrl);
   return {
-    changed: changed || !existingAccessControl,
+    changed: changed || !existingPolicies,
```

Comment:
Given that you've already returned above if `existingPolicies` is truthy, isn't this always true?

```suggestion
    changed: true,
```